src/ConnectionPool.jl

```diff
@@ -70,17 +85,34 @@ mutable struct Connection{T <: IO}
     peerport::UInt16 # debug only
     localport::UInt16 # debug only
     io::T
+    clientconnection::Bool
     buffer::IOBuffer
-    sequence::Int
-    writecount::Int
-    writebusy::Bool
-    writedone::Condition
-    readcount::Int
-    readbusy::Bool
-    readdone::Condition
+    sequence::Threads.Atomic{Int}
```

Comment:
I noticed after saying that that this Atomic seems correct to keep (or you could make it non-atomic and protected by the writebusy-lock, but then the ordering would be imposed by `startwrite` instead of `Transaction`, which might not be what you wanted)

Reply:
Yeah, I'm refactoring right now and I left this as Atomic.